From af13b4922b057b4ebc7e2315a6f666ecb65890e4 Mon Sep 17 00:00:00 2001
From: David Sterba <dsterba@suse.cz>
Date: Wed, 30 Jul 2014 00:58:37 +0200
Subject: btrfs: fix error labels in init_btrfs_fs

btrfs_interface_init rarely fails but we could leak the prelim_ref slab.

Signed-off-by: David Sterba <dsterba@suse.cz>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index b1d2a42..b915d77 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -1999,11 +1999,11 @@ static int __init init_btrfs_fs(void)
 
 	err = btrfs_prelim_ref_init();
 	if (err)
-		goto free_prelim_ref;
+		goto free_delayed_ref;
 
 	err = btrfs_interface_init();
 	if (err)
-		goto free_delayed_ref;
+		goto free_prelim_ref;
 
 	btrfs_init_lockdep();
 
-- 
cgit v0.10.1

